### PR TITLE
Flexible num reducers

### LINKED ISSUE #55: Make number of reducers flexible
Make the maximum number of reducers flexible, instead of hardcoding the limit to 4. Allows for more interesting tests.
(See motivation in eecs485staff/p5-search-engine#550)

### PR DESCRIPTION
**Pull request by MattyMay**
Add optional `-numReduceTasks` argument. Closes #55.

### CODE REVIEW

**Review by melodell — COMMENTED** [review]
LGTM! (but I wrote some of this code so wait for another approval lol)

**Review by awdeorio — APPROVED** [review]
LGTM

**Pull request merged**
